server/weaverbird/backends/sql_translator/steps/todate.py

```diff
@@ -0,0 +1,57 @@
+from distutils import log
+
+from weaverbird.backends.sql_translator.steps.utils.query_transformation import (
+    build_selection_query,
+    complete_fields,
+    snowflake_date_format,
+)
+from weaverbird.backends.sql_translator.types import (
+    SQLPipelineTranslator,
+    SQLQuery,
+    SQLQueryDescriber,
+    SQLQueryRetriever,
+)
+from weaverbird.pipeline.steps import ToDateStep
+
+
+def translate_todate(
+    step: ToDateStep,
+    query: SQLQuery,
+    index: int,
+    sql_query_retriever: SQLQueryRetriever = None,
+    sql_query_describer: SQLQueryDescriber = None,
+    sql_translate_pipeline: SQLPipelineTranslator = None,
+) -> SQLQuery:
+    query_name = f'TODATE_STEP_{index}'
+
+    log.debug(
+        "############################################################"
+        f"query_name: {query_name}\n"
+        "------------------------------------------------------------"
+        f"step.name: {step.name}\n"
+        f"step.column: {step.column}\n"
+        f"step.format: {step.format}\n"
+        f"query.transformed_query: {query.transformed_query}\n"
+        f"query.metadata_manager.tables_metadata: {query.metadata_manager.tables_metadata}\n"
+        f"query.metadata_manager.query_metadata: {query.metadata_manager.query_metadata}\n"
+    )
+    step.format = snowflake_date_format(step.format)
+    # we change the column type
+    query.metadata_manager.change_type(step.column, 'date')
+    new_query = SQLQuery(
+        query_name=query_name,
+        transformed_query=f"""{query.transformed_query}, {query_name} AS"""
+        f""" (SELECT {complete_fields(columns=[step.column], query=query)},"""
+        f""" TRY_TO_DATE({step.column}{step.format}) AS {step.column}"""
```

Comment:
Au final, en discutant de la gestion des erreurs avec @f-bb-toucantoco, on s'est dit qu'il ne vaut mieux pas utiliser les fonctions TRY_XXX (TRY_DATE, TRY_CAST, ...) et plutôt laisser l'utilisateur se prendre des erreurs plutôt que d'afficher un NULL, dont il ne comprendrait pas la provenance. 
```suggestion
        f""" TO_DATE({step.column}{step.format}) AS {step.column}"""
```